### PR TITLE
Issue/171/react on errors correctly

### PR DESCRIPTION
**Pull request by dbetka**

### CODE REVIEW

File: src/api/real/event-controller.js [hunk 1]
```diff
@@ -68,14 +65,17 @@ export const eventController = {
...
-            reject(new ErrorMessage(ERRORS.collectPoint));
-          }
-        })
-        .catch(catchConnectionError(reject));
+        .then(data => apiResponseService.takeOverResponse({
+          data,
+          resolve,
+          reject,
+          defaultError: ERRORS.collectPoint,
+          errors: [
```
[the remark this answers is not in the document]
Comment (dbetka) [reply src/api/real/event-controller.js:73]: There is problem with that solution - each validateCode has own message for a specific endpoint. So all requests have different messages for the same validateCodes.

File: src/utils/api-response-service.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * @param errors - example:
+ *   errors: [
+ *     [errorCode1, errorMessage1],
+ *     [errorCode2, errorMessage2],
+ *   ],
+ */
+export const apiResponseService = {
+  takeOverResponse ({
+    data,
```
[the remark this answers is not in the document]
Comment (dbetka) [reply src/utils/api-response-service.js:14]: Good point i will fix it

File: src/utils/request.js [hunk 3]
```diff
@@ -1,6 +1,36 @@
...
+  return new Promise((resolve, reject) => {
+    method({
+      url,
+      data,
+      ...fetchOptions,
+    })
+      .then(response => apiResponseService.takeOverResponse({
+        response,
+        onSuccess: responseData => {
+          delete responseData.error;
```
Comment (Bedpaw) [inline src/utils/request.js:22]: Line 22 delete responseData.error;
I think it should belong to apiResponseService

Comment (dbetka) [reply src/utils/request.js:22]: Fix

**Pull request merged**